test/testutil/fakeoctopusserver.go

```diff
@@ -55,6 +55,8 @@ type MockHttpServer struct {
 	// and go panics and kills the process, so we find out about it, but this is a bit
 	// less confusing to troubleshoot
 	pendingMsgCount int32
+
+	Closed bool
```

Comment:
This isn't bulletproof, but adding in the ability to "shut down" the fake HTTP server and fake survey Asker means that in many cases a test failure doesn't deadlock anymore.

Deadlock is not so bad (the test fails, so you go fix it), but the problem is, when go detects a deadlock it hard panics and you don't get to see any of the stdout or stderr so you don't know what happened leading up to the test failure. Usually there's an assert "expected value blah, got blah2" that you'd like to see